src/GraphQL/Instrumentation/ApolloTracingExtensions.cs

```diff
@@ -65,20 +64,5 @@ public static void EnrichWithApolloTracing(this ExecutionResult result, DateTime
 
             return trace;
         }
-
-        private static IEnumerable<object> ConvertPath(IEnumerable<string> stringPath)
-        {
-            foreach (var step in stringPath)
-            {
-                if (int.TryParse(step, out var arrayIndex))
-                {
-                    yield return arrayIndex;
```

Comment:
It's funny, but in this place, the behavior matches the specification, converting the string back to a number.